File: JINSEOP/프로그래머스_WEEK13/단속 카메라.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<vector<int>> routes) {
    int answer = 1;

    sort(routes.begin(), routes.end());
    int curr_camera = routes[0][1];

    for (int i = 1; i < routes.size(); i++) {
        if (curr_camera < routes[i][0]) {
            answer++;
            curr_camera = routes[i][1];
        }
        if (routes[i][1] <= curr_camera)
            curr_camera = routes[i][1];
    }

    return answer;
}

File: DONGUK/프로그래머스_WEEK_13/단속 카메라.py
from collections import deque

def solution(routes):
    answer = 0
    routes.sort(key = lambda x: (x[1], x[0]))
    
    q = deque(routes)
    
    while q:
        top = q.popleft()
        cur_start, cur_end = top[0], top[1]
        answer += 1
        
        while q and cur_end >= q[0][0]:
            q.popleft()
                
    return answer